### PR TITLE
Use 'path' arg for NWBHDF5IO if file is local

### PR DESCRIPTION
**Pull request by alejoe91**
Fix needed for local NWB files after #2294

### CODE REVIEW

File: src/spikeinterface/extractors/nwbextractors.py [hunk 1]
```diff
@@ -112,10 +112,6 @@ def _read_hdf5_file(
         assert file_path is not None, "file_path must be specified when using stream_mode='remfile'"
         rfile = remfile.File(file_path)
         hdf5_file = h5py.File(rfile, "r")
-
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/nwbextractors.py:115]: I don't think we need to remove this case as it works for the current file in the CI.

File: src/spikeinterface/extractors/nwbextractors.py [hunk 2]
```diff
@@ -174,14 +170,20 @@ def read_nwbfile(
...
 
-    hdf5_file = _read_hdf5_file(
-        file_path=file_path,
-        file=file,
-        stream_mode=stream_mode,
-        cache=cache,
-        stream_cache_path=stream_cache_path,
-    )
-    io = NWBHDF5IO(file=hdf5_file, mode="r", load_namespaces=True)
+    if file_path is not None and stream_mode is None:
```
Comment (h-mayorquin) [inline src/spikeinterface/extractors/nwbextractors.py:173]: But for your specific file this general fix will determine the behavior as `_read_hdf5_file` will not be called.

**Review by h-mayorquin — APPROVED** [review]
As mentioned the file in the CI is working. So there is something about your file or your environment. I would like to find this out but given that this work for both your local file and the CI one this seems like a more robust solution.

File: src/spikeinterface/extractors/nwbextractors.py (hunk 1 above)
Comment (h-mayorquin) [reply src/spikeinterface/extractors/nwbextractors.py:115]: I actually would like to avoid removing this, can we do that?

**Comment by alejoe91** [comment]
Turns out it went away after uninstalling and reinstalling `pynwb`

**Pull request closed without merging**